core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/storage/ScanSourceFileResolver.scala

```diff
@@ -0,0 +1,37 @@
+package edu.uci.ics.texera.workflow.common.storage
+
+import java.nio.file.{Files, Path, Paths}
+import edu.uci.ics.amber.engine.common.storage.DatasetFileDocument
+import scala.util.{Failure, Success, Try}
+
+object ScanSourceFileResolver {
+
+  case class SourceFileNotFound(message: String) extends Exception(message)
+
+  /**
+    * Attempts to resolve the given fileName to either a local file path or a DatasetFileDocument.
+    *
+    * @param fileName the name of the file to resolve
+    * @return a tuple of (Option[Path], Option[DatasetFileDocument])
+    */
+  def resolve(fileName: String): (Option[String], Option[DatasetFileDocument]) = {
```

Comment:
Use Either[String, Document], otherwise both can be None legally.

Reply:
Changed